### PR TITLE
PROG-39044 upgrade MongoDB version to latest

### PR DESCRIPTION
**Pull request by shauncmar**
Story: [PROG-39044](https://newforma.atlassian.net/browse/PROG-39044)

Context: 

From Pascal: 

> fyi, I had an issue to deploy cloud-documentdb-schema-tool in prod today where creating an index is failing with a not master error when the cluster has a primary and at least 1 replica instance (dev and stage only have a primary instance, no replica). I had to rollback the common-lambda package update in cloud-documentdb-schema-tool for now. It seems to be caused by a breaking changes in the mongodb npm package between version 3.5.9 and 3.6.2 used by common-lambda. It is using 3.6.2 since v30.0.0 but query and crud statements don't look to be affected as the cloud-workflow-items repository is using common-lambda >= v30.0.0 and its acceptance tests in prod are green. I found https://github.com/mongodb/node-mongodb-native/releases/tag/v3.6.3, they mention the not master is a regression bug. Probably should update common-lambda to latest mongodb v3.6.4 to fix our issue.

### CODE REVIEW

**Review by nkayurov-newforma — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.3.0 
+- update mongodb version to 3.6.4
```
Comment (nkayurov-newforma) [inline CHANGELOG.md]: I think it was done in the previous version. Here we just updated @types/mongodb package.

**Review by diaamostafa — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (pgxtreme) [reply CHANGELOG.md]: It's only upgrading the devDependencies that are used only by this repo, I don't feel these changes are really necessary. The text your';e quoting from me was about common-lambda, not this repository.

**Review by pgxtreme — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (shauncmar) [reply CHANGELOG.md]: hmmm so we do not also need to update this repository to ensure the fix is propagated forward?

**Review by Ta3k — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (pgxtreme) [reply CHANGELOG.md]: You don't need to update this repository, mongodb is a peer dependency, it means it's going to use the mongodb package used by the repo where mongo-cursor-pagination-alt is added as a dependency. The work is already done, might as well merge it but it's not gonna do anything at the common-lambda level.

**Pull request merged**